src/components/TaskItem.tsx

```diff
@@ -3,10 +3,15 @@ import { TaskItemType } from '../types/TaskItemType';
 
 interface TaskItemProps {
   taskData: TaskItemType;
+  onEdit: (taskTitle: string, taskId: number) => void;
 }
 
-function TaskItem({ taskData }: TaskItemProps) {
-  const [isChecked, setIsChecked] = useState(taskData.done);
+function TaskItem({ taskData, onEdit }: TaskItemProps) {
+  const [isChecked, setIsChecked] = useState<boolean>(taskData.done);
+
+  const handleEdit = () => {
+    onEdit(taskData.title, taskData.id);
+  };
 
   return (
     <li className="todo stack-small">
```

Comment:
Not related exactly to this changes, but Ive noticed that this <li> here should not be placed since the element in the end is not rendered inside an list `<ul>` or `<ol>`. I believe you can change here to just a div, or simplify the whole structure.